### ISSUE TITLE
Bug: Neither SA1304 nor SA1307 is reported for public readonly fields

### ISSUE THREAD

**Issue by test20150828** [DotNetAnalyzers/StyleCopAnalyzers#1306]
Hi,

It seems like both analyzers rely to each other to report a warning and no warning is generated at all. Here's a simple example (StyleCop.Analyzers 1.0.0.8):

```C#
#pragma warning disable SA1401 // Fields must be private
public class Class1
{
	// no warning here
	public readonly int sa1304or1307shouldBeReportedHere;
}
```

**Comment by pdelvo** [comment]
Yes this looks like a :bug: to me. The [source code](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1304NonPrivateReadonlyFieldsMustBeginWithUpperCaseLetter.cs#L82) for SA1304 clearly says that SA1307 should take precedence.

Do you want to work on this yourself or should someone take a look at this?

**Comment by test20150828** [comment]
No, I just wanted to report a bug. However, I think the fix is just to remove `|| declaration.Modifiers.Any(SyntaxKind.ReadOnlyKeyword)` in SA1307AccessibleFieldsMustBeginWithUpperCaseLetter.cs, line 63 (provided that SA1307 should have priority in this case).

**Comment by pdelvo** [comment]
Yes this should fix it. Im going to grab this

**Comment by sharwell** [comment]
Thanks for the :bug: report @test20150828!
